components/channel_header_dropdown/menu_items/close_message/close_message.js

```diff
@@ -62,21 +62,40 @@ export default class CloseDirectChannel extends React.PureComponent {
             },
         } = this.props;
 
-        const category = Constants.Preferences.CATEGORY_DIRECT_CHANNEL_SHOW;
+        let name;
+        let category;
+        if (channel.type === Constants.DM_CHANNEL) {
+            category = Constants.Preferences.CATEGORY_DIRECT_CHANNEL_SHOW;
+            name = channel.teammate_id;
+        } else {
+            category = Constants.Preferences.CATEGORY_GROUP_CHANNEL_SHOW;
+            name = channel.id;
+        }
 
-        savePreferences(currentUser.id, [{user_id: currentUser.id, category, name: channel.teammate_id, value: 'false'}]);
+        savePreferences(currentUser.id, [{user_id: currentUser.id, category, name, value: 'false'}]);
 
         browserHistory.push(`/${currentTeam.name}/channels/${redirectChannel}`);
     }
 
     render() {
         const {id, channel} = this.props;
+
+        if (channel.type === Constants.DM_CHANNEL) {
```

Comment:
We only swap the text between these two cases, so we could factor that out and return the same component with different text.
```javascript
let text;
if (channel.type === DM_CHANNEL) {
    text = ...;
} else if (channel.type === GM_CHANNEL) {
    text = ...;
}

return (
    <Menu.ItemAction
        show={channel.type === DM_CHANNEL || channel.type === GM_CHANNEL}
        text={text}
    />
);
```